src/main.rs

```diff
@@ -45,48 +50,38 @@ fn publish(docs_path: &str,
     git::force_push(docs_path, remote.trim(), &refspec, dry_run)
 }
 
-fn execute(args: &ArgMatches) -> Result<i32, FatalError> {
-    let properties: Table = try!(config::parse_config());
-
-    // Verify the TOML configuration.
-    let valid_keys = vec![config::DOCS_PATH,
-                          config::COMMIT_MESSAGE,
-                          config::SIGN_COMMIT,
-                          config::PUSH_REMOTE,
-                          config::PUSH_BRANCH];
-
-    for key in properties.keys() {
-        if !valid_keys.contains(&key.as_ref()) {
-            println!("Unknown config key \"{}\" found for [package.metadata.sphinx]",
-                     key);
-            return Ok(109);
-        }
-    }
+fn execute(args: &ArgMatches, cargo_config: &CargoConfig) -> Result<i32, FatalError> {
+    try!(cargo_config.configure_shell(args.occurrences_of("verbose") as u32,
+                                      Some(args.is_present("quiet")),
+                                      &args.value_of("color").map(|s| String::from(s))));
+
+    let config: Config = try!(Config::from("Cargo.toml"));
 
     // Find parameters or use defaults.
     let dry_run = args.is_present("dry-run");
 
     let docs_path = args.value_of("docs-path")
-        .or_else(|| config::get_str(&properties, config::DOCS_PATH))
+        .or_else(|| config.get_str(config::DOCS_PATH))
         .unwrap_or("docs");
 
     let push = args.is_present("push");
 
     let commit_msg = args.value_of("commit-message")
-        .or_else(|| config::get_str(&properties, config::COMMIT_MESSAGE))
+        .or_else(|| config.get_str(config::COMMIT_MESSAGE))
         .unwrap_or("(cargo-sphinx) Generate docs.");
-    let sign = args.is_present("sign") ||
-               config::get_bool(&properties, config::SIGN_COMMIT).unwrap_or(false);;
+
+    let sign = args.is_present("sign") || config.get_bool(config::SIGN_COMMIT).unwrap_or(false);
 
     let push_remote = args.value_of("push-remote")
-        .or_else(|| config::get_str(&properties, config::PUSH_REMOTE))
+        .or_else(|| config.get_str(config::PUSH_REMOTE))
         .unwrap_or("origin");
+
     let push_branch = args.value_of("push-branch")
-        .or_else(|| config::get_str(&properties, config::PUSH_BRANCH))
+        .or_else(|| config.get_str(config::PUSH_BRANCH))
         .unwrap_or("gh-pages");
 
     // Generate and publish documentation.
-    try!(build(docs_path, dry_run));
+    try!(build(docs_path, &mut *cargo_config.shell(), dry_run));
```

Comment:
Honestly though, I looked at [RefMut](https://doc.rust-lang.org/core/cell/struct.RefMut.html) and was terrified for a second until I saw `deref` and remembered that `*` is just an alias for that.